Create unit tests for ssg.id_translate

jan-cerny opened the pull request:
#### Description:
Create unit tests for ssg.id_translate

#### Rationale:
Increases testing of build scripts.

jan-cerny commented:
I have rebased this PR on the top of the latest upstream branch to include the fix for the duplicate CCE problems.

matejak commented on tests/unit/ssg-module/test_id_translate.py:
```diff
@@ -21,3 +26,172 @@ def test__split_namespace():
     assert n == "emptynamespace"
+
+
+def test_tagname_to_abbrev():
```
Wouldn't it be better to define a mapping `tagname->expected abbreviation, and just iterate over the pairs?

matejak commented on tests/unit/ssg-module/data/draft_oval.xml:
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0"?>
```
What about editing the OVAL, so it contains less verbose and more versatile data? For example, the ID could be rewritten to `RULE_ID`, and things that are not tested and not required by the build system code for smooth finish could get removed from the XML file.

matejak commented on tests/unit/ssg-module/test_id_translate.py:
```diff
@@ -21,3 +26,172 @@ def test__split_namespace():
+    return ET.parse(draft_oval_path).getroot()
+
+
+def test_idtranslator_translate_oval(idtranslator, oval_tree):
```
I see that this test takes an OVAL tree, and "translates" it, whatever it means.
I would expect this test to focus on expected differences between original and translated (or expected lack of differences). Instead, the test is mainly about making sure that the translated tree has certain properties, and it does so without referencing the original.

matejak reviewed (changes requested):
Thank you for the PR, I have some questions and suggestions though.
My main concern is that these tests are not really readable, and that they may be fragile.

It may be that there is no other way, but we need to look into that.

jan-cerny commented:
I have refactor the test using a dictionary and I have removed some data from OVAL to reduce test data input and I have changed the unit tests for `translate()` and I have rebased on the latest upstream master.

jan-cerny commented:
@matejak I have add a test that uses the xml diff using the xmldiff library.

Also, I have rebased this PR on the top of the latest upstream master branch.

jan-cerny commented:
Based on discussion with the team I have removed the tests that check
expected properties of the output XML tree from the translate method,
because we will prefer the test based on xmldiff that allows us to
compare the expected diff with the actual diff.

jan-cerny commented:
@matejak Are there any further changes needed?

jan-cerny commented on tests/unit/ssg-module/test_id_translate.py:
```diff
@@ -21,3 +29,214 @@ def test__split_namespace():
+                f'{o}:criteria/{o}:criterion[1]'),
+            name='test_ref',
+            value='oval:ssg-test_kerberos_disable_no_keytab:tst:1')
+    }
```
@matejak said that it would be better if the expected items are removed from the diff and finally it would be checked if the diff is empty.

jan-cerny commented:
I have removed the expected items from the diff

matejak commented:
Thank you for the PR with those tests, I appreciate them, and we can think of any possible improvements of their readability later - it is important to start with something solid first, and this is what the PR brings.
